File: iosApp/iosAppTests/UIComponent/DebugMenu/DebugMenu/DebugMenuStateRobot.swift
#if DEBUG
import ComposeApp

final class DebugMenuStateRobot: BaseStateRobot<DebugMenuUiState, DebugMenuEvent> {

    override func defaultState() -> DebugMenuUiState {
        DebugMenuUiState(
            entries: [
                DebugMenuEntry(
                    id: "feature-flags",
                    title: "Feature Flags",
                    subtitle: "Inspect and override remote flags",
                    target: FeatureFlagsDebugScreen.shared
                ),
                DebugMenuEntry(
                    id: "build-config",
                    title: "Build Config",
                    subtitle: "Current environment, SDK versions, and build metadata",
                    target: BuildConfigDebugScreen.shared
                ),
            ],
            eventSink: createEventSink()
        )
    }

    func stateWithNoEntries() -> DebugMenuUiState {
        DebugMenuUiState(entries: [], eventSink: createEventSink())
    }
}
#endif

File: features/mobile/debug-menu/impl/presentation/src/androidDeviceTest/kotlin/com/mockdonalds/app/features/mobile/debugmenu/presentation/DebugMenuStateRobot.kt
package com.mockdonalds.app.features.mobile.debugmenu.presentation

import com.mockdonalds.app.core.test.StateRobot
import com.mockdonalds.app.features.mobile.debugmenu.api.navigation.BuildConfigDebugScreen
import com.mockdonalds.app.features.mobile.debugmenu.api.navigation.FeatureFlagsDebugScreen

class DebugMenuStateRobot : StateRobot<DebugMenuUiState, DebugMenuEvent>() {

    override fun defaultState() = DebugMenuUiState(
        entries = listOf(
            DebugMenuEntry(
                id = "feature-flags",
                title = "Feature Flags",
                subtitle = "Inspect and override remote flags",
                target = FeatureFlagsDebugScreen,
            ),
            DebugMenuEntry(
                id = "build-config",
                title = "Build Config",
                subtitle = "Current environment, SDK versions, and build metadata",
                target = BuildConfigDebugScreen,
            ),
        ),
        eventSink = createEventSink(),
    )

    fun stateWithNoEntries() = defaultState().copy(
        entries = emptyList(),
        eventSink = createEventSink(),
    )
}
